Pledge-Based Saturation Limit Under a Closed System/README.md

```diff
@@ -0,0 +1,252 @@
+  
+![logic2](https://drive.google.com/uc?export=view&id=1j45ZEEhqwDDJoXcVxs4p1NqixegXGbmM)  
+  
+## **_Sybil Attack_**
+
+We can calculate the total pledge to conduct a Sybil attack using the following equation:
+
+![cost](https://drive.google.com/uc?export=view&id=1cqZJ7y9oBi1bA6rOYWzlseUAIyOH3avg)
+
+We consider a scenario where some pools turn adversarial (**_k<sub>adversarial</sub>_**), and we can give the best scenario for the attackers by assuming that their pools are fully saturated. We can then plot the total pledge to conduct a Sybil attack as well as **_k<sub>adversarial</sub>_** vs the number of pools (**_k_**).
+
+![costfuncpools](https://drive.google.com/uc?export=view&id=1lW93crB3YIFeOd4Y8iG0fxkmKHZKlrgr)
+
+We found that under this CIP, the number of pools needed to attack the ecosystem (**_k<sub>adversarial</sub>_**) increases linearly with the total number of pools (left figure). This observation is in line with the expectation that an ecosystem's security should become more robust as the number of pools increases. Unfortunately, this expectation is **not true** for the protocol currently implemented in Cardano. In addition, notice the existence of a minimum in the plot (right figure) at around 1600 pools. The total pledge to conduct a Sybil attack at this minimum is 45 M ADA in pledge (when **_n_** = 15) and at least 600 pools turning adversarial. The total pledge to conduct a Sybil attack increases as the number of pools either decreases or increases from this minimum. For epoch 321 at **_n_** = 15, the total pledge to conduct a Sybil attack is 52 M ADA in pledge and at least 1108 adversarial pools. We dare say that the total pledge to conduct a Sybil attack under the current protocol is 0 ADA in pledge and only 177 adversarial pools. This number of adversarial pools needed to attack the ecosystem under the current protocol does not change even if the total number of pools increases, unless the saturation limit (68 M) is decreased. These pools can easily get saturated by implementing lucrative (and maybe fake) ISPOs. This mechanism of stake centralization had already been demonstrated during the Sundaeswap ISPO which involved 98 pools that were at or near saturation.

```

Comment:
I was hoping to find a justification to a concern I've got since I read the abstract but didn't: it seems to me that the proposal does not actual make it particularly more difficult for adversarial to conduct a Sybil attack? Indeed, while the proposals tends to increase the number of pools, by defining an optimal pledge and making the saturation more dynamic, it doesn't prevent a pool operator from running many pools. Actually, it seems that the proposal even incentivizes pool owner with large pledge to split their pledge into many pools. This has been a recurring discussion on [CIP-0050?](https://github.com/cardano-foundation/CIPs/pull/242) which is tricky to address. All-in-all, I fail to see how forcing the system into many more pools necessarily help decentralization, without ensuring that those pools are owned by distinct owners. 
In the current scheme, owners are incentivized to keep high pledges in their pools because it generates slightly better rewards and thus, hopefully attract more delegators. I'd be curious to see numbers of the total pledge in the current state of affairs for conducting a Sybil Attack (So, take the first ~21-23 biggest pools and sum-up their pledge). How would that compare to this proposal?

Reply:
Indeed!

As you may have read in my recent comments, I formulated the proposal under the assumption that 1 hardware = 1 pool which I recently found to be not true. This CIP will only prosper when 1 hardware can be limited to just a very few pools - this CIP will not prosper in the current state of affairs.

Further, I noted in my recent comments that the weakest link in the ecosystem appears to be the fact that 1 hardware can proliferate an unlimited number of pools using different IP addresses and without incurring any additional cost. This appears to be the weakest link in the ecosystem, and any attempt to improve decentralization will always carry this weakest link, as you have noted that it is also true for [CIP-50](https://github.com/cardano-foundation/CIPs/pull/242).

It may be necessary to address this weakest link first before we attempt to improve decentralization. I will propose that instead of using a public IP address, there may be a need to require to have their IP address associated with only 1 MAC address. Although this will still allow an operator to generate multiple pools via the use of virtual machines, this will severely limit their capability. Here are some calculations I made for 1 MAC = 1 IP:

A PC with
- 64 cores (highest count to date)
- 256 GB RAM (8 slots, 32 GB per slot)
- 100% CPU core allocation
- 100% memory allocation

can accommodate 20 of 12 GB VMs only (calculated using this [website](https://wintelguy.com/vmcalc.pl)). Therefore, under this CIP, the 1108 adversarial pools that are needed to attack the current state of the network will need 554 of this high-end PC. Further, 1108 VMs on a cloud is computationally expensive.

Increasing the requirement from 12 GB RAM to 24, 32, 64, and even 128 GB will further diminish their capability. Further, additional RAM requirements will not necessarily generate significant energy use.

As for the calculation, if you mean 21-22 biggest groups, the current pledge is indeed bigger compared to what was shown in CIP-45 calculations. But also CIP-45 is not setting any optimal pledge just yet, but I do recognize that there may be a need to re-define how optimal pledge is computed so that small pools will stay competitive.

Reply:
Okay. I may have found the answer to your concern.

1) remove the penalty factor.
2) To keep the premise of a closed system, each pool with excess pledge will have k >1. Say a pool with 3x pledge will be counted as 3 in the counting of k.
3) Excess pledge will be incentivized with more "luck" for example(?) This way operators will be encouraged to keep their pledges in one pool rather than splitting them.
4) Rewards for delegators will still be fixed at 4.5%, subject to pool performance so they need not flock to a near-saturated pool. If they are delegated to a pool that is nonpeformant, their reward will decline.